### PR TITLE
Wizard: Add additional information for an activation key in Review step

### LINKED ISSUE #822: Add additional information for a selected activation key
When user selects an activation key for registration, wizard should show additional information connected to the activation key.

The displayed fields will help user determine if they selected the correct key.

This will be needed for the `Review` and `Registration` (https://github.com/RedHatInsights/image-builder-frontend/issues/603) steps.

### PR DESCRIPTION
**Pull request by regexowl**
Fixes #822. This adds component `ActivationKeyInformation` with additional information for an activation key on a Registration tab of Review step. The added fields are:
- Name
- Role
- SLA
- Usage
- Additional repositories

Organization ID is added to the Activation key popover.

Function `getActivationKey` was added to `api` to get an additional information about activation key from RHSM api based on the name of the key.

### CODE REVIEW

File: src/Components/CreateImageWizard/formComponents/ActivationKeyInformation.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+                      aria-label="Additional repositories table"
+                      variant="compact"
+                    >
+                      <Thead>
+                        <Tr>
+                          <Th>Name</Th>
+                        </Tr>
+                      </Thead>
+                      <Tbody data-testid="additional-repositories-table">
+                        {additionalRepositories?.map((r, ri) => (
```
Comment (lucasgarfield) [inline src/Components/CreateImageWizard/formComponents/ActivationKeyInformation.js:112]: This is personal preference but maybe call these (repo, index)? For me, there is some mental overhead required to convert `r` to `repo`. But we don't have a style guide that specifies either way and this is also fine if you prefer it.

**Review by lucasgarfield — CHANGES_REQUESTED** [review]
This is looking great!!!

I think we should probably add test coverage for this before we merge it... The new API call can be mocked, take a look at how we use `jest.spyOn().mockImplementation()` in `ImagesTable.test.js`.

File: src/Components/CreateImageWizard/formComponents/ActivationKeyInformation.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            {usage || 'Not defined'}
+          </TextListItem>
+          <TextListItem component={TextListItemVariants.dt}>
+            Additional <br /> repositories:
+            <Popover
+              bodyContent={
+                <TextContent>
+                  <Text>
+                    The core repositories for your operating system version are
+                    always enabled and do not need to be explicitally added to
```
[the remark this answers is not in the document]
Comment (regexowl) [reply src/Components/CreateImageWizard/formComponents/ActivationKeyInformation.js]: Fixed!

File: src/Components/CreateImageWizard/formComponents/ActivationKeyInformation.js (hunk 1 above)
Comment (regexowl) [reply src/Components/CreateImageWizard/formComponents/ActivationKeyInformation.js]: Fair, updated for better readability :)

File: src/Components/CreateImageWizard/formComponents/ReviewStep.js [hunk 3]
```diff
@@ -85,6 +88,7 @@ const ReviewStep = () => {
       (async () => {
         const userData = await insights?.chrome?.auth?.getUser();
         const id = userData?.identity?.internal?.org_id;
+        setOrgId(id);
```
[the remark this answers is not in the document]
Comment (regexowl) [reply src/Components/CreateImageWizard/formComponents/ReviewStep.js]: Updated

**Comment by regexowl** [comment]
@lucasgarfield Thanks for the comments! I've updated the code and added test coverage for the shown additional information.

File: src/Components/CreateImageWizard/formComponents/ActivationKeyInformation.js [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+              >
+                <HelpIcon />
+              </Button>
+            </Popover>
+          </TextListItem>
+          <TextListItem
+            component={TextListItemVariants.dd}
+            className="pf-u-display-flex pf-u-align-items-flex-end"
+          >
+            {additionalRepositories?.length === 0 ? (
```
Comment (lucasgarfield) [inline src/Components/CreateImageWizard/formComponents/ActivationKeyInformation.js:93]: If `additionalRepositories` is `undefined` for whatever reason, then we will render the `<Popover>` component... which I don't think we want. What if this was something like: `additionalRepositories?.length > 0 ? (<Popover ... />) : ('None')`? That way, we only render the popover with the additional repositories table if we are 100% sure there are additional repositories.

File: src/Components/CreateImageWizard/formComponents/ReviewStep.js [hunk 5]
```diff
@@ -315,22 +317,34 @@ const ReviewStep = () => {
...
+                            {getState()?.values?.[
+                              'subscription-organization-id'
+                            ] !== undefined ? (
+                              `Your organization's ID is ${
+                                getState()?.values?.[
+                                  'subscription-organization-id'
+                                ]
+                              }`
+                            ) : (
+                              <Spinner />
```
Comment (lucasgarfield) [inline src/Components/CreateImageWizard/formComponents/ReviewStep.js:330]: If you disable the line of code that sets `'subscription-organization-id'` so you can see what happens if it is `undefined`, the popover looks like this:

spinner

What if we tried something like `Your organization's ID is <Spinner size="sm" />.`?

File: src/Components/CreateImageWizard/formComponents/ActivationKeyInformation.js (hunk 4 above)
Comment (regexowl) [reply src/Components/CreateImageWizard/formComponents/ActivationKeyInformation.js]: Fixed :+1:

**Review by lucasgarfield — APPROVED** [review]

**Pull request merged**